mars/dataframe/indexing/iloc.py

```diff
@@ -0,0 +1,244 @@
+
+        op = DataFrameIlocSetItem(indexes=_extend_indexes(indexes), value=value, object_type=ObjectType.dataframe)
+        ret = op(self._obj)
+        self._obj.data = ret.data
+
+
+def _extend_indexes(indexes):
+    # Extend the indexes to a 2-D indexes
+    if isinstance(indexes, tuple):
+        if len(indexes) == 1:
+            return indexes + (slice(None, None, None),)
+        else:
+            return indexes
+    else:
+        return (indexes, slice(None, None, None))
+
+
+class DataFrameIlocGetItem(DataFrameOperand, DataFrameOperandMixin):
+    _op_type_ = OperandDef.DATAFRAME_ILOC_GETITEM
+
+    _indexes = ListField('indexes')
+
+    def __init__(self, indexes=None, gpu=False, sparse=False, object_type=None, **kw):
+        super(DataFrameIlocGetItem, self).__init__(_indexes=indexes,
+                                                   _gpu=gpu, _sparse=sparse,
+                                                   _object_type=object_type, **kw)
+
+    @property
+    def indexes(self):
+        return self._indexes
+
+    def __call__(self, df):
+        from ...tensor.indexing.getitem import _getitem
+        from ...tensor.datasource.empty import empty
+
+        # Note [Fancy Index of Numpy and Pandas]
+        #
+        # The numpy and pandas.iloc have different semantic when processing fancy index:
+        #
+        # >>> np.ones((3,3))[[1,2],[1,2]]
+        # array([1., 1.])
+        #
+        # >>> pd.DataFrame(np.ones((3,3))).iloc[[1,2],[1,2]]
+        #    1    2
+        # 1  1.0  1.0
+        # 2  1.0  1.0
+        #
+        # Thus, we processing the index along two axis of DataFrame seperately.
+
+        tensor0 = _getitem(empty(df.shape[0]), self.indexes[0])
+        tensor1 = _getitem(empty(df.shape[1]), self.indexes[1])
+
+        if isinstance(self.indexes[0], Integral) or isinstance(self.indexes[1], Integral):
+            # NB: pandas only compresses the result to series when index on one of axis is integral
+            if isinstance(self.indexes[1], Integral):
+                shape = tensor0.shape
+                dtype = df.dtypes.iloc[self.indexes[1]]
+                index_value = indexing_index_value(df.index_value, self.indexes[0])
+            else:
+                shape = tensor1.shape
+                dtype = find_common_type(df.dtypes.iloc[self.indexes[1]].values)
+                index_value = indexing_index_value(df.columns, self.indexes[1])
+            self._object_type = ObjectType.series
+            return self.new_series([df], shape=shape, dtype=dtype, index_value=index_value)
+        else:
+            return self.new_dataframe([df], shape=tensor0.shape + tensor1.shape, dtypes=df.dtypes.iloc[self.indexes[1]],
+                                      index_value=indexing_index_value(df.index_value, self.indexes[0]),
+                                      columns_value=indexing_index_value(df.columns, self.indexes[1], store_data=True))
+
+    def on_output_modify(self, new_output):
```

Comment:
`on_input_modify` and `on_output_modify` should appear together..

Reply:
After more detailed investgation I think it's not trivial to get the same behavior with pandas about the _view_ behavior of `DataFrame.iloc`, see the example in `FIXME`.

I have removed the `on_output_modify` part.

Reply:
Fine, we could consider view behavior of DataFrame at some point later.